From 87292f0f279197c11111ebe65fc4f0fa05e393b8 Mon Sep 17 00:00:00 2001
From: Sergio Prado <sergio.prado@toradex.com>
Date: Thu, 12 Aug 2021 06:06:42 -0300
Subject: [PATCH] onigmo: fix cross-compiling issue

Building failing with the folowing error due to missing sysroot
configuration:

regint.h:225:10: fatal error: stdlib.h: No such file or directory
  225 | #include <stdlib.h>

Signed-off-by: Sergio Prado <sergio.prado@toradex.com>
---
 cmake/onigmo.cmake | 16 +++++-----------
 1 file changed, 5 insertions(+), 11 deletions(-)

diff --git a/cmake/onigmo.cmake b/cmake/onigmo.cmake
index efa596443b7f..0b47abc1eb9f 100644
--- a/cmake/onigmo.cmake
+++ b/cmake/onigmo.cmake
@@ -28,26 +28,20 @@ endif()
 # Onigmo (UNIX)
 # =============
 if(FLB_SMALL)
-ExternalProject_Add(onigmo
-  BUILD_IN_SOURCE TRUE
-  EXCLUDE_FROM_ALL TRUE
-  SOURCE_DIR ${ONIGMO_SRC}
-  INSTALL_DIR ${ONIGMO_DEST}
-  CONFIGURE_COMMAND ./configure ${AUTOCONF_HOST_OPT} --host=${HOST} --with-pic --disable-shared --enable-static --prefix=${ONIGMO_DEST}
-  CFLAGS=-std=gnu99\ -Wall\ -pipe\ -Os\ -g0\ -s\ -fno-stack-protector\ -fomit-frame-pointer\ -DNDEBUG\ -U_FORTIFY_SOURCE
-  BUILD_COMMAND $(MAKE) CC=${DEPS_C_COMPILER}
-  INSTALL_COMMAND $(MAKE) DESTDIR= install)
+  set(CFLAGS "$ENV{CFLAGS} -std=gnu99 -Wall -pipe -Os -g0 -s -fno-stack-protector -fomit-frame-pointer -DNDEBUG -U_FORTIFY_SOURCE")
 else()
+  set(CFLAGS "$ENV{CFLAGS} -std=gnu99 -Wall -pipe -g3 -O3 -funroll-loops")
+endif()
+
 ExternalProject_Add(onigmo
   BUILD_IN_SOURCE TRUE
   EXCLUDE_FROM_ALL TRUE
   SOURCE_DIR ${ONIGMO_SRC}
   INSTALL_DIR ${ONIGMO_DEST}
   CONFIGURE_COMMAND ./configure ${AUTOCONF_HOST_OPT} --host=${HOST} --with-pic --disable-shared --enable-static --prefix=${ONIGMO_DEST}
-  CFLAGS=-std=gnu99\ -Wall\ -pipe\ -g3\ -O3\ -funroll-loops
+  CFLAGS=${CFLAGS}
   BUILD_COMMAND $(MAKE) CC=${DEPS_C_COMPILER}
   INSTALL_COMMAND $(MAKE) DESTDIR= install)
-endif()
 
 # Onigmo (Windows)
 # ================
-- 
2.25.1

